src/imp/libc/fs/syscalls.rs

```diff
@@ -123,6 +123,28 @@ pub(crate) fn openat(
     oflags: OFlags,
     mode: Mode,
 ) -> io::Result<OwnedFd> {
+    // Work around https://sourceware.org/bugzilla/show_bug.cgi?id=17523
+    // If we have a getrandom() symbol, then we know we have a glibc that
+    // has fixed the bug.  This can be removed when Rust bumps its minimum
+    // glibc version.
+    if !crate::imp::is_atleast_glibc_2_25() {
```

Comment:
We could also skip this check if `O_TMPFILE` isn't in `oflags` perhaps?

Reply:
Yes, that would be good to do.